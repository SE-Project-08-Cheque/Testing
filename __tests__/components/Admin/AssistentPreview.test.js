// import dependencies
import React from "react";

// import API mocking utilities from Mock Service Worker
import { rest } from "msw";
import { setupServer } from "msw/node";

// import renderer for take snapshots
import renderer from "react-test-renderer";

// import react-testing methods
import { render, cleanup, fireEvent } from "@testing-library/react";

// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";

// import react dom
import ReactDOM from "react-dom";

// import the component for testing
import AssistantPreview from "../../../components/Admin/AssistentPreview";



afterEach(cleanup);



const server = setupServer(
  rest.get("/employees/2", (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json({
        firstname: "chandima",
        lastname: "Amarasena",
        email: "chandima334@gmail.com",
      })
    );
  })
);

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<AssistantPreview data={{ firstname: "chandima" }} />, div);
});

it("renders assistant preview correctly", () => {
  const { getByTestId } = render(
    <AssistantPreview
      data={{
        firstname: "Chandima",
        lastname: "Amarasena",
        email: "chandima334@gmail.com",
      }}
    />
  );
  expect(getByTestId("firstname")).toHaveTextContent("Chandima");
  expect(getByTestId("lastname")).toHaveTextContent("Amarasena");
  expect(getByTestId("email")).toHaveTextContent("chandima334@gmail.com");
});

it("edit button works correctly", () => {
  const { queryByTestId } = render(
    <AssistantPreview
      data={{
        firstname: "Chandima",
        lastname: "Amarasena",
        email: "chandima334@gmail.com",
      }}
    />
  );

 
  expect(queryByTestId("firstname")).toBeInTheDocument();
  expect(queryByTestId("lastname")).toBeInTheDocument();
  expect(queryByTestId("email")).toBeInTheDocument();

  expect(queryByTestId("firstname_input")).not.toBeInTheDocument();
  expect(queryByTestId("lastname_input")).not.toBeInTheDocument();
  expect(queryByTestId("email_input")).not.toBeInTheDocument();

  fireEvent.click(queryByTestId("edit_button")) 

  expect(queryByTestId("firstname")).not.toBeInTheDocument();
  expect(queryByTestId("lastname")).not.toBeInTheDocument();
  expect(queryByTestId("email")).not.toBeInTheDocument();

  expect(queryByTestId("firstname_input")).toBeInTheDocument();
  expect(queryByTestId("lastname_input")).toBeInTheDocument();
  expect(queryByTestId("email_input")).toBeInTheDocument();

  fireEvent.change(queryByTestId("firstname_input"),{ target: { value: 'chandima' } })
  fireEvent.change(queryByTestId("lastname_input"),{ target: { value: 'amarasena' } })
  fireEvent.change(queryByTestId("email_input"),{ target: { value: 'chandima@gmail.com' } })

  fireEvent.click(queryByTestId("show_button")) 

  fireEvent.click(queryByTestId("submit_button")) 

  
});

it("matches snapshot", () => {
  const tree = renderer.create(
    <AssistantPreview
      data={{
        placename: "Galle",
        distance: "20",
        timeToReach: "30",
        description: "no description",
        methods: ["bus"],
      }}
    />
  );

  expect(tree).toMatchSnapshot();
});
