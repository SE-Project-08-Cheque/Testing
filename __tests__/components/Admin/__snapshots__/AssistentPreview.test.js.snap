// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-mccp0p"
>
  <form
    action="submit"
  >
    <div
      className="chakra-stack css-1n2jfo2"
    >
      <div
        className="css-gmuwbf"
      >
        <span
          className="chakra-avatar css-1ruhx4w"
        >
          <div
            aria-label="Profile photo"
            className="chakra-avatar__initials css-0"
          >
            Pp
          </div>
        </span>
      </div>
      <p
        className="chakra-text css-504jus"
      >
        <span
          className="chakra-badge css-ifsvay"
        >
          First name
        </span>
      </p>
      <p
        className="chakra-text css-1n49pl7"
        data-testid="firstname"
      />
      <hr
        aria-orientation="horizontal"
        className="chakra-divider css-1upb9tn"
        role="separator"
      />
      <p
        className="chakra-text css-504jus"
      >
        <span
          className="chakra-badge css-ifsvay"
        >
          Last name
        </span>
      </p>
      <p
        className="chakra-text css-1n49pl7"
        data-testid="lastname"
      />
      <hr
        aria-orientation="horizontal"
        className="chakra-divider css-1upb9tn"
        role="separator"
      />
      <p
        className="chakra-text css-504jus"
      >
        <span
          className="chakra-badge css-ifsvay"
        >
          Email
        </span>
      </p>
      <p
        className="chakra-text css-1n49pl7"
        data-testid="email"
      />
      <button
        className="chakra-button css-oqfhsh"
        data-testid="edit_button"
        onClick={[Function]}
        type="button"
      >
        Edit details
      </button>
    </div>
  </form>
</div>
`;
